File: src/Application Specific/Classes/Entity Collections/ContractsOriginal/ContractOriginalSiteAsset/ContractOriginalSiteAsset.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace ContractOriginalSiteAssets

        Public Class ContractOriginalSiteAsset

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

#End Region

#Region "ContractSiteAsset Properties"


            Private _ContractSiteAssetID As Integer = 0
            Public ReadOnly Property ContractSiteAssetID() As Integer
                Get
                    Return _ContractSiteAssetID
                End Get
            End Property
            Public WriteOnly Property SetContractSiteAssetID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ContractSiteAssetID", Value)
                End Set
            End Property


            Private _ContractSiteID As Integer = 0
            Public ReadOnly Property ContractSiteID() As Integer
                Get
                    Return _ContractSiteID
                End Get
            End Property
            Public WriteOnly Property SetContractSiteID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ContractSiteID", Value)
                End Set
            End Property


            Private _AssetID As Integer = 0
            Public ReadOnly Property AssetID() As Integer
                Get
                    Return _AssetID
                End Get
            End Property
            Public WriteOnly Property SetAssetID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_AssetID", Value)
                End Set
            End Property


            Private _PricePerVisit As Double = 0
            Public ReadOnly Property PricePerVisit() As Double
                Get
                    Return _PricePerVisit
                End Get
            End Property
            Public WriteOnly Property SetPricePerVisit() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PricePerVisit", Value)
                End Set
            End Property


            Private _PrimaryAsset As Boolean = 0
            Public ReadOnly Property PrimaryAsset() As Boolean
                Get
                    Return _PrimaryAsset
                End Get
            End Property
            Public WriteOnly Property SetPrimaryAsset() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PrimaryAsset", Value)
                End Set
            End Property


            Private _Type As String = ""
            Public ReadOnly Property Type() As String
                Get
                    Return _Type
                End Get
            End Property
            Public WriteOnly Property SetType() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Type", Value)
                End Set
            End Property

            Private _Product As String = ""
            Public ReadOnly Property Product() As String
                Get
                    Return _Product
                End Get
            End Property
            Public WriteOnly Property SetProduct() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Product", Value)
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/ContractsOriginal/ContractOriginalSiteAsset/ContractOriginalSiteAsset.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ContractOriginalSiteAssets
    {
        public class ContractOriginalSiteAsset
        {
            private DataTypeValidator _dataTypeValidator;

            public ContractOriginalSiteAsset()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            public object SetContractSiteAssetID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ContractSiteAssetID", value);
                }
            }

            private int _ContractSiteID = 0;

            public int ContractSiteID
            {
                get
                {
                    return _ContractSiteID;
                }
            }

            public object SetContractSiteID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ContractSiteID", value);
                }
            }

            private int _AssetID = 0;

            public int AssetID
            {
                get
                {
                    return _AssetID;
                }
            }

            public object SetAssetID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AssetID", value);
                }
            }

            private double _PricePerVisit = 0;

            public double PricePerVisit
            {
                get
                {
                    return _PricePerVisit;
                }
            }

            public object SetPricePerVisit
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PricePerVisit", value);
                }
            }

            private bool _PrimaryAsset = Conversions.ToBoolean(0);

            public bool PrimaryAsset
            {
                get
                {
                    return _PrimaryAsset;
                }
            }

            public object SetPrimaryAsset
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PrimaryAsset", value);
                }
            }

            private string _Type = "";

            public string Type
            {
                get
                {
                    return _Type;
                }
            }

            public object SetType
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Type", value);
                }
            }

            private string _Product = "";

            public string Product
            {
                get
                {
                    return _Product;
                }
            }

            public object SetProduct
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Product", value);
                }
            }
        }
    }
}